### PR TITLE
Add gpu

### PR DESCRIPTION
**Pull request by EuanPyle**
Allows user to specify GPU in the exact same manner as MotionCor2 in rln e.g. user provides a string with colon separated GPU IDs: 0:1:2:3

### CODE REVIEW

**Comment by EuanPyle** [comment]
Will merge if you think the setup makes sense @alisterburt

File: lil_aretomo/aretomo.py [hunk 1]
```diff
@@ -23,7 +24,8 @@ def run_aretomo_alignment(
         nominal_rotation_angle: Optional[float] = None,
         n_patches_xy: Optional[Tuple[int, int]] = (5, 4),
         thickness_for_alignment: Optional[float] = 1500,
-        correct_tilt_angle_offset: Optional[bool] = False	
+        correct_tilt_angle_offset: Optional[bool] = False,
+        gpu_id: Optional[str] = None
```
Comment (alisterburt) [inline lil_aretomo/aretomo.py:28]: would prefer that our interface take a tuple of ints and convert -> don't want to impose RELION-isms on our Python API
```suggestion
        gpu_id: Optional[Tuple[int, ...]] = None	
```

What do you think?

Comment (alisterburt) [reply lil_aretomo/aretomo.py:28]: would need a few other changes to make this work ofc

Comment (EuanPyle) [reply lil_aretomo/aretomo.py:28]: Sweet, will have tuple of ints as input and ask for rln to pass the info as a tuple of ints!

Comment (alisterburt) [reply lil_aretomo/aretomo.py:28]: test out what the CLI parser wants for tuple of ints before asking just to make sure :) I think it's just space separated ints `0 1 2 3`

Comment (EuanPyle) [reply lil_aretomo/aretomo.py:28]: Typer won't accept an ellipsis as a type hint which is annoying, may have to keep it as a str?

Comment (EuanPyle) [reply lil_aretomo/aretomo.py:28]: This is an issue in tomo_PP not lil_aretomo

Comment (EuanPyle) [reply lil_aretomo/aretomo.py:28]: Could have: relion takes string (e.g. 0:1:2:3), tomo_PP takes this string and removes the ":", passes to lil_AT as a tuple. Does this make sense?

Comment (alisterburt) [reply lil_aretomo/aretomo.py:28]: cool okay, can you rename to gpu_ids and specify the expected formatting in the docstring? :)

Comment (alisterburt) [reply lil_aretomo/aretomo.py:28]: will open an issue at typer

Comment (alisterburt) [reply lil_aretomo/aretomo.py:28]: wait a sec, what if we type as `List[int]` - should work right?

Comment (EuanPyle) [reply lil_aretomo/aretomo.py:28]: I'll give that a go!

Comment (EuanPyle) [reply lil_aretomo/aretomo.py:28]: No, still doesn't like it. I can't pass a list on the CLI, only thing that works is List[int] which only allows me to pass one value. If I change to List[int,...] I get a slightly different error with:  TypeError: Parameters to generic types must be types. Got Ellipsis.

**Comment by EuanPyle** [comment]
All working now. Lil_AT take tuple, tomo_PP takes a str and converts to tuple! Will merge lil_AT now as it's what we agreed, but tomo_PP changes need to be approved

**Pull request merged**